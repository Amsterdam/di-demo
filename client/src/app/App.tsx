import React from 'react';
import styled from 'styled-components';
import { Route, BrowserRouter as Router } from 'react-router-dom';
import { GlobalStyle, ThemeProvider, Row, themeColor } from '@datapunt/asc-ui';
import theme from '@services/theme';
import * as AscLocal from '@components/LocalAsc/LocalAsc';
import Footer from '@components/Footer/Footer';
import AppRoutes from './AppRoutes';
import ScrollToTop from '@components/ScrollToTop/ScrollToTop';

const App: React.FC<unknown> = () => (
    <ThemeProvider overrides={theme}>
        <Bg>
            <GlobalStyle />
            <StyledRow>
                <AscLocal.Header fullWidth={false} tall homeLink={AppRoutes.HOMEPAGE.path} />
            </StyledRow>
            <Router>
                <ScrollToTop />
                {Object.keys(AppRoutes).map(key => {
                    const { path, component, exact } = AppRoutes[key];
                    return <Route key={key} path={path} component={component} exact={exact} />;
                })}
            </Router>
            <Footer />
        </Bg>
    </ThemeProvider>
);

const StyledRow = styled(Row)`
    background-color: ${themeColor('tint', 'level1')};
`;

const Bg = styled.div`
    background-color: ${themeColor('tint', 'level3')};
`;

export default App;
