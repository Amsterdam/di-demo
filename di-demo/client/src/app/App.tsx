import * as React from 'react';

const App: React.FC<unknown> = () => <h1>App!</h1>;

export default App;
