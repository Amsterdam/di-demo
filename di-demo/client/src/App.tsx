import React, { useEffect, useContext } from "react";
import styled from "styled-components";
import "./App.css";
import { PageWrapper } from "./AppStyle";

const MainContainerStyle = styled.div`
  background-color: #f2f5f8;
`;

const LinkListStyle = styled.div`
  position: absolute;
  top: 326px;
  left: 0;
  right: 0;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const LinkStyle = styled.a`
  // border: solid 1px grey;
  width: 314px;
  margin-bottom: 20px;
  height: 53px;
`;

const Spacer = styled.div<{ height: number }>`
  height: ${({ height }) => height}px;
`;

const themeName = "amsterdam";
const App: React.FC = () => {

  useEffect(() => {
    (async () => {
      const response = await fetch("http://localhost:8000/stats");
      const data = await response.json();
      console.log("got", data);
    })();
  }, []);

  return (
    <MainContainerStyle>
      <PageWrapper>
        <img
          alt="Home"
          src={`assets/theme/${themeName}/home.png`}
          height="1068"
          width="1400"
          decoding="async"
        />
        <LinkListStyle>
          <LinkStyle href={`mijnamsterdam/${themeName}`}>
            {/* Mijn Amsterdam */}
          </LinkStyle>
          <LinkStyle href={`openstad/${themeName}`}>
            {/* Open stad */}
          </LinkStyle>
          <LinkStyle href={`halloijburg/${themeName}`}>
            {/* Hallo IJburg */}
          </LinkStyle>
          <LinkStyle href={`alchoolkopen/${themeName}`}>
            {/* Alchool kopen */}
          </LinkStyle>
          <LinkStyle href={`rommelmelden/${themeName}`}>
            {/* Rommel melden */}
          </LinkStyle>
          <Spacer height={41} />
          <LinkStyle href={`kortingkrijgen/${themeName}`}>
            {/* Korting krijgen */}
          </LinkStyle>
        </LinkListStyle>
      </PageWrapper>
    </MainContainerStyle>
  );
};

export default App;
