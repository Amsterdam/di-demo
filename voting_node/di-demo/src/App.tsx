import React from "react";
import styled from "styled-components";
import "./App.css";
import { PageWrapper } from "./AppStyle";

const MainContainerStyle = styled.div`
  background-color: #F2F5F8;
`;

const LinkListStyle = styled.div`
  position: absolute;
  top: 326px;
  left: 0;
  right: 0;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const LinkStyle = styled.a`
  // border: solid 1px grey;
  width: 314px;
  margin-bottom: 20px;
  height: 53px;
`;

const Spacer = styled.div<{ height: number }>`
  height: ${({ height }) => height}px;
`;

const App: React.FC = () => {
  return (<MainContainerStyle>

    <PageWrapper>
        <img alt="Home" src="https://s3.invisionapp-cdn.com/storage.invisionapp.com/screens/files/398337770.png?x-amz-meta-iv=3&amp;response-cache-control=max-age%3D2419200&amp;x-amz-meta-ck=6c5383e9abd1130f655ddd13ea0bf466&amp;AWSAccessKeyId=AKIAJFUMDU3L6GTLUDYA&amp;Expires=1580515200&amp;Signature=TZOGJH%2FKxtyL4QWifGzDYYfnTy0%3D" height="1068" width="1400" decoding="async"/>
        <LinkListStyle>
          <LinkStyle href="mijnamsterdam">{/* Mijn Amsterdam */}</LinkStyle>
          <LinkStyle href="openstad">{/* Open stad */}</LinkStyle>
          <LinkStyle href="halloijburg">{/* Hallo IJburg */}</LinkStyle>
          <LinkStyle href="alchoolkopen">{/* Alchook kopen */}</LinkStyle>
          <LinkStyle href="rommelmelden">{/* Rommel melden */}</LinkStyle>
          <Spacer height={41} />
          <LinkStyle href="kortingkrijgen">{/* Korting krijgen */}</LinkStyle>
        </LinkListStyle>
    </PageWrapper>
  </MainContainerStyle>
  );
};

export default App;
