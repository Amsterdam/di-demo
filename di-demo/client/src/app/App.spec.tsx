import React from 'react';
import App from './App';
import { render } from '@testing-library/react';

describe('App component', () => {
    it('should match snapshot', () => {
        const { asFragment } = render(<App />);
        expect(asFragment()).toMatchSnapshot();
    });
});
