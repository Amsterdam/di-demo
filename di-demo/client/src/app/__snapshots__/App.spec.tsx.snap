// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<DocumentFragment>
  <h1>
    App!
  </h1>
</DocumentFragment>
`;
